hw/ip/entropy_src/data/entropy_src.hjson

```diff
@@ -239,208 +239,256 @@ param_list: [
                    first and only test run, or when this mode is programmed by firmware.
-                   The default value is 32
+                   This register must be written before this module is enabled, and
+                   once written, will only reflect a value lower than the current value.
                 '''
-          resval: "0x0020"
+          resval: "0xFFFF"
         }
       ]
     },
     { name: "ADAPTP_HI_THRESHOLDS",
       desc: "Adaptive proportion test high thresholds register",
       swaccess: "rw",
-      hwaccess: "hro",
+      hwaccess: "hrw",
+      hwext: "true",
+      hwqe:  "true",
       regwen:   "REGWEN",
+      tags: [// Exclude from writes to these field because they are one-way updates
+             "excl:CsrAllTests:CsrExclWrite"]
       fields: [
         { bits: "15:0",
           name: "FIPS_ADAPTP_HI_THRESH",
           desc: '''This is the threshold size for the adaptive proportion health test.
                    This value is used in normal mode when entropy is being tested in
                    FIPS/CC compliance mode.
-                   The default value is 598.
+                   This register must be written before this module is enabled, and
+                   once written, will only reflect a value lower than the current value.
                 '''
-          resval: "0x0256"
+          resval: "0xFFFF"
         }
         { bits: "31:16",
           name: "BYPASS_ADAPTP_HI_THRESH",
           desc: '''This is the threshold size for the adaptive proportion health test
                    running in bypass mode. This mode is active after reset for the
                    first and only test run, or when this mode is programmed by firmware.
-                   The default value is .375 of the FIPS threshold.
+                   This register must be written before this module is enabled, and
+                   once written, will only reflect a value lower than the current value.
                 '''
-          resval: "0x00e0"
+          resval: "0xFFFF"
         }
       ]
     },
     { name: "ADAPTP_LO_THRESHOLDS",
       desc: "Adaptive proportion test low thresholds register",
       swaccess: "rw",
-      hwaccess: "hro",
+      hwaccess: "hrw",
+      hwext: "true",
+      hwqe:  "true",
       regwen:   "REGWEN",
+      tags: [// Exclude from writes to these field because they are one-way updates
+             "excl:CsrAllTests:CsrExclWrite"]
       fields: [
         { bits: "15:0",
           name: "FIPS_ADAPTP_LO_THRESH",
           desc: '''This is the threshold size for the adaptive proportion health test.
                    This value is used in normal mode when entropy is being tested in
                    FIPS/CC compliance mode.
-                   The default value is 426.
+                   This register must be written before this module is enabled, and
+                   once written, will only reflect a value lower than the current value.
```

Comment:
does this one need to guard in the opposite direction? Ie only go higher?

Reply:
Good catch, I noticed this already when addressing @cdgori wording comment. It is wrong is a number of places, hopefully all corrected now.